pylint/config/arguments_manager.py

```diff
@@ -33,7 +34,6 @@ def __init__(self) -> None:
         self._arg_parser = argparse.ArgumentParser(
             prog="pylint",
             usage="%(prog)s [options]",
```

Comment:
Behaviour changed here after `3.7`. Couldn't see this in the docs but by comparing `argparse.py` I finally figured it out.

Since we register most checkers in `argparse` now there is no ambiguity anymore, or at least it is figured out by `argparse`, so this should be okay.

I have tested with both `3.7` and `3.10` locally.